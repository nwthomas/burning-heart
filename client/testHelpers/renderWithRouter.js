import React from "react";
import { NativeRouter as Router } from "react-router-native";
import { render } from "react-testing-library";
import { createMemoryHistory } from "history";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history
  };
};

export default renderWithRouter;
