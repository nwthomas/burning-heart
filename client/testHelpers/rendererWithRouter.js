import React from "react";
import { NativeRouter as Router } from "react-router-native";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(<Router history={history}>{ui}</Router>),
    history
  };
};

export default rendererWithRouter;
