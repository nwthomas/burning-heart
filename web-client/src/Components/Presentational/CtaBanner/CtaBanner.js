import React from "react";

const CtaBanner = () => {
  return (
    <div className="cta__banner">
      <p className="cta__banner__text">
        Burning Heart is a donation app for people who like to give what they
        can, when they can.{" "}
        <a className="cta__banner__link" href="#">
          Sign up here.
        </a>
      </p>
    </div>
  );
};

export default CtaBanner;
