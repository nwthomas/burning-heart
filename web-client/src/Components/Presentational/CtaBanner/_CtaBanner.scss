.cta__banner {
  width: 100%;
  padding: 10px 3%;
  background: black;
  display: flex;
  justify-content: center;
  align-items: center;

  .cta__banner__text {
    color: white;
    line-height: 1.5;
    font-size: 1.4rem;
    text-align: center;

    .cta__banner__link {
      color: #ffb2b3;

      &:hover {
        text-decoration: underline;
      }
    }
  }
}
