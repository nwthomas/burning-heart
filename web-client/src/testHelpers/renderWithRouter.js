import React from "react";
import { Router } from "react-router-dom";
import { render } from "react-testing-library";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../store";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(
      <Router history={history}>
        <Provider store={store}>{ui}</Provider>
      </Router>
    ),
    history
  };
};

export default renderWithRouter;
