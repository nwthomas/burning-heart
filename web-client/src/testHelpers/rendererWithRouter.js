import React from "react";
import { Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../store";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <Router history={history}>
        <Provider store={store}>{ui}</Provider>
      </Router>
    ),
    history
  };
};

export default rendererWithRouter;
