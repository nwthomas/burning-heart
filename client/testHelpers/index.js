import renderWithRouter from "./renderWithRouter";
import rendererWithRouter from "./rendererWithRouter";

export { renderWithRouter, rendererWithRouter };
