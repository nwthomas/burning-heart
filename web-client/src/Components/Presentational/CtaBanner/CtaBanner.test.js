import React from "react";
import "react-testing-library/cleanup-after-each";
import "jest-dom/extend-expect";
import { renderWithRouter, rendererWithRouter } from "../../../testHelpers";

import CtaBanner from "./CtaBanner";

describe("<CtaBanner />", () => {
  it("should render the CtaBanner components without crashing", () => {
    const helpers = renderWithRouter(<CtaBanner />);
  });

  it("should match the snapshot of CtaBanner", () => {
    const tree = rendererWithRouter(<CtaBanner />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
