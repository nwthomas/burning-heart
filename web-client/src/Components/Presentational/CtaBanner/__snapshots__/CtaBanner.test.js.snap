// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CtaBanner /> should match the snapshot of CtaBanner 1`] = `
<div
  className="cta__banner"
>
  <p
    className="cta__banner__text"
  >
    Burning Heart is a donation app for people who like to give what they can, when they can.
     
    <a
      className="cta__banner__link"
      href="/signup"
      onClick={[Function]}
    >
      Sign up here.
    </a>
  </p>
</div>
`;
